botnim/benchmark/assistant_loop.py

```diff
@@ -64,34 +81,190 @@ def assistant_loop(client: OpenAI, assistant_id, question=None, thread=None, not
+                )
+                
+                # Use context-specific settings if available
+                num_results = arguments.get('num_results', 
+                                         context_config.get('max_num_results', 3))
+                
+                # Log the tool call parameters
+                logger.info(f"Calling run_query with query: {arguments['query']}, num_results: {num_results}")
+                
+                output = run_query(
+                    environment=environment,
+                    bot_name=bot_name,
+                    context_name=context_name,
+                    query=arguments['query'],
+                    num_results=num_results,
+                    format="text"
+                )
+                
+                # Log the output
+                logger.info(f"Tool output: {output}")
+            
+            # Handle all non-search tools with OpenAPI
+            else:
+                # For non-search tools, use OpenAPI
+                if tool.function.name == 'DatasetDBQuery':
+                    # Special case for DatasetDBQuery - already set page_size earlier
+                    pass
+                
+                # Call get_openapi_output for all non-search tools
+                try:
+                    if openapi_spec is not None:
+                        output = get_openapi_output(openapi_spec, tool.function.name, arguments)
+                        
+                        if tool.function.name == 'DatasetInfo':
+                            # Special case for DatasetInfo
+                            output = get_dataset_info_cache(arguments, output)
+                    else:
+                        output = f"Error: OpenAPI spec not provided for tool {tool.function.name}"
+                except Exception as e:
+                    logger.error(f"Error calling {tool.function.name}: {e}")
+                    output = f"Error: {str(e)}"
+            
+            if output is not None:
+                # Log tool output
+                with open(log_file, 'a', encoding='utf-8') as f:
+                    f.write(f"\nTool Output:\n  Tool ID: {tool.id}\n  Content:\n")
+                    if isinstance(output, str):
+                        f.write(f"    {output}\n")
+                    else:
+                        # For dictionary/list outputs, format them nicely
+                        output_str = json.dumps(output, ensure_ascii=False, indent=4)
+                        # Add indentation to each line
+                        formatted_output = "\n".join(f"    {line}" for line in output_str.split("\n"))
+                        f.write(f"{formatted_output}\n")
+                
+                tool_outputs.append(dict(
+                    tool_call_id=tool.id,
+                    output=json.dumps(output, ensure_ascii=False, indent=2)
+                ))
+
+        if tool_outputs:
+            run = client.beta.threads.runs.submit_tool_outputs(
```

Comment:
why did you change this part of the code?

Reply:
to prevent submitting empty empty tool outputs, reverted.